datasync/downloader/peer.go

```diff
@@ -210,6 +220,28 @@ func (p *peerConnection) FetchReceipts(request *fetchRequest) error {
 	return nil
 }
 
+// FetchStakingInfo sends a staking information retrieval request to the remote peer.
+func (p *peerConnection) FetchStakingInfo(request *fetchRequest) error {
+	// Sanity check the protocol version
+	if p.version < 65 {
+		panic(fmt.Sprintf("staking info fetch [klay/65+] requested on klay/%d", p.version))
```

Comment:
If a peer does not support protocol/65, it panics. Is this right?

Reply:
@aidan-kwon @kjhman21 @ethan-kr 
I have checked the logic again. `FetchingStakingInfo` method is called to only idle peers which are satisfying the version match as well. So, it should be fine. Please take another look.

- `fetchParts`
```
// numTotalPeers means the number of peers satisfying the protocol requirement.
idlePeers, numTotalPeers := idlePeers()
pendCount := pending()

for _, peer := range idlePeers {
...
if err := fetch(peer, request); err != nil {
...
```

- `idlePeers`
```
func (ps *peerSet) idlePeers(minProtocol, maxProtocol int, idleCheck func(*peerConnection) bool, throughput func(*peerConnection) float64) ([]*peerConnection, int) {
	ps.lock.RLock()
	defer ps.lock.RUnlock()

	idlePeers, numTotalPeers := make([]*peerConnection, 0, len(ps.peers)), 0
	tps := make([]float64, 0, len(ps.peers))
	for _, p := range ps.peers {
		if p.version >= minProtocol && p.version <= maxProtocol {
			if idleCheck(p) {
				idlePeers = append(idlePeers, p)
				tps = append(tps, throughput(p))
			}
			numTotalPeers++
		}
	}
	// sort peers in the descending order of throughput
	sortPeers := &peerThroughputSort{idlePeers, tps}
	sort.Sort(sortPeers)
	return sortPeers.p, numTotalPeers
}
```